File: clients/java/src/main/java/com/priostack/acn/ConnectResult.java
package com.priostack.acn;

import com.fasterxml.jackson.databind.JsonNode;
import java.util.List;

/**
 * Result of {@link ACNClient#connect}.
 *
 * <p>The connect envelope's {@code data} uses PascalCase Go field names
 * ({@code SessionID}, {@code ResolvedAccount}, ...); this client maps them to
 * idiomatic Java accessors. All other tools use lowercase keys.
 */
public final class ConnectResult {

    private final String sessionId;
    private final String resolvedAccount;
    private final List<String> grantedCapabilities;
    private final List<String> grantedContextRights;
    private final String boundSpace;
    private final int resolvedMaxTokens;
    private final JsonNode raw;

    public ConnectResult(
            String sessionId,
            String resolvedAccount,
            List<String> grantedCapabilities,
            List<String> grantedContextRights,
            String boundSpace,
            int resolvedMaxTokens,
            JsonNode raw) {
        this.sessionId = sessionId;
        this.resolvedAccount = resolvedAccount;
        this.grantedCapabilities = grantedCapabilities;
        this.grantedContextRights = grantedContextRights;
        this.boundSpace = boundSpace;
        this.resolvedMaxTokens = resolvedMaxTokens;
        this.raw = raw;
    }

    /** The session id (from PascalCase {@code SessionID}), captured on the client. */
    public String sessionId() {
        return sessionId;
    }

    public String resolvedAccount() {
        return resolvedAccount;
    }

    public List<String> grantedCapabilities() {
        return grantedCapabilities;
    }

    public List<String> grantedContextRights() {
        return grantedContextRights;
    }

    public String boundSpace() {
        return boundSpace;
    }

    public int resolvedMaxTokens() {
        return resolvedMaxTokens;
    }

    public JsonNode raw() {
        return raw;
    }

    @Override
    public String toString() {
        return "ConnectResult{sessionId="
                + sessionId
                + ", resolvedAccount="
                + resolvedAccount
                + ", resolvedMaxTokens="
                + resolvedMaxTokens
                + '}';
    }
}

File: clients/php/src/Result/ConnectResult.php
<?php

declare(strict_types=1);

namespace Priostack\Acn\Result;

/**
 * Result of {@see \Priostack\Acn\Client::connect()}.
 *
 * NOTE: the {@code connect} envelope {@code data} is the only one that uses
 * PascalCase Go field names ({@code SessionID}, {@code ResolvedAccount}, ...).
 * This factory maps them onto lower-camelCase PHP properties; every other tool
 * uses lowercase keys.
 */
final class ConnectResult
{
    /**
     * @param list<string>         $grantedCapabilities
     * @param list<string>         $grantedContextRights
     * @param array<string, mixed> $raw
     */
    public function __construct(
        public readonly string $sessionId,
        public readonly string $resolvedAccount,
        public readonly array $grantedCapabilities,
        public readonly array $grantedContextRights,
        public readonly string $boundSpace,
        public readonly int $resolvedMaxTokens,
        public readonly array $raw = [],
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromData(array $data): self
    {
        return new self(
            sessionId: (string) ($data['SessionID'] ?? ''),
            resolvedAccount: (string) ($data['ResolvedAccount'] ?? ''),
            grantedCapabilities: array_values((array) ($data['GrantedCapabilities'] ?? [])),
            grantedContextRights: array_values((array) ($data['GrantedContextRights'] ?? [])),
            boundSpace: (string) ($data['BoundSpace'] ?? ''),
            resolvedMaxTokens: (int) ($data['ResolvedMaxTokens'] ?? 0),
            raw: $data,
        );
    }
}
